tsc/meetings/2019-03-28.md

```diff
@@ -0,0 +1,150 @@
+a startup. Andrew Grimberg is taking over his responsibilities and is already
+up-to-speed. It is envisioned that Andrew will attend the TAC and CI working
+group meetings, but likely not the project TSC meetings.
+
+4) Multi-Res Grid
+
+Ken has received a comprehensive response from Autodesk answering the TSC
+questions regarding Autodesk's proposal for introducing the Multi-Res Grid to
+OpenVDB. He will be sending it around to the group shortly for discussion in the
+next meeting. First impressions look promising.
+
+5) Out-of-Core Read-only Grid
+
+Ken has a working implementation of an out-of-core VDB grid. It is read-only,
+supports random access with a pre-defined fixed memory footprint and can be
+built from a standard out-of-core VDB grid. A fully out-of-core VDB grid
+requires half the memory footprint of a standard VDB grid in out-of-core mode.
+Savings are also possible through not supporting very old file formats. Ken
+plans to release at Weta and then to look to introduce to the OpenVDB codebase.
+Some concerns from Peter and Dan about introducing a new Grid structure, but
+keen on understanding more of the implementation details.
+
+6) Instantiation / Compile-times
+
+Compilation of applications and plugins built against OpenVDB is slow and uses a
+lot of memory, which we wish to address. Peter wants to consider baking the
+configuration of the grid so as to improve compilation times and suggests asking
+users whether they ever change the grid configuration. Ken mentioned that he
+knows people change the size of the leaf node for GPU and other uses, but
+unaware of many people changing the configuration of the tree. He wants to find
+out if this is a consideration for Autodesk's Multi-Res Grid. Potential
+investigation around where the compilation time is going could help direct our
+efforts. Explicit template specialization also worth looking into.
+
+7) Tools vs Grid Methods
+
+Over time, there has been a gradual migration towards using free functions in
+the tools directory instead of adding to the Grid and Tree classes. Question of
+whether it's a good time to introduce a policy around this and to look at
+removing some of the Grid and Tree methods. Ken proposed that the first step is
+to look at all the methods and confirm whether they can currently be removed or
+whether they need new access functions to be able to remove them. This change is
+also expected to improve compilation times. Removing virtual functions needs to
+be done in a major release, so they would need to be marked deprecated in a
+prior release.
+
+8) Release Schedule
+
+Peter proposes we do a new release in the next few weeks. It was decided that
+the next release should be 6.1.0, as there is a lot of new functionality and
+some of the changes appear that they may affect the API. Unanimous vote in
+favour. Potentially hold off to ensure we can include CMake changes. Peter
+proposes adding static casts to fix the precision warnings so that we can make
```

Comment:
Just to clarify, I'm proposing to add static casts only temporarily, with a more permanent fix to follow, most likely in the 7.0 release.

Reply:
Yes, that's what I understood. I just updated the wording to make that clear.